ts/parts/Chart.ts

```diff
@@ -478,6 +478,24 @@ extend(Chart.prototype, /** @lends Highcharts.Chart.prototype */ {
             userOptions.series = null as any;
             options = merge(defaultOptions, userOptions); // do the merge
 
+            const optionsChart: Highcharts.ChartOptions = options.chart || {},
+                panning = optionsChart && optionsChart.panning,
+                verticalPanningEnabled = panning && /y/.test(panning.type);
+
+            // Automatically disable the yAxis start and endOnTick,
+            // when vertical panning is enabled
+            options.yAxis = options.yAxis = splat(options.yAxis || {}).map(function (
+                yAxisOptions: Highcharts.YAxisOptions
+            ): Highcharts.YAxisOptions {
+                return merge(
+                    yAxisOptions,
+                    (verticalPanningEnabled ? {
+                        startOnTick: false,
+                        endOnTick: false
+                    } : null)
+                );
+            });
+
```

Comment:
I think it is bad practice to modify the options set based on other options (although we do it in other places too). I think this code block should be rewritten to do the test in the places where `startOnTick` and `endOnTick` are actually used, in `Axis.ts`. You could create a private function that tests for `hasVerticalPanning`, to use in the conditions checking for `startOnTick` or `endOnTick`. Then refactor when you see more economic ways of doing it.

For example, in the top of `setTickPositions`, we could do:
```js
var .....,
    verticalPanning = hasVerticalPanning(chart),
    startOnTick = !verticalPanning && options.startOnTick,
    endOnTick = !verticalPanning && options.endOnTick
```